fastcrawler/test_utils/endpoints.py

```diff
@@ -0,0 +1,182 @@
+    """
+    return Response(text=text, status_code=405, headers=headers)
+
+
+class BaseEndpoint:
+    """A base class that represents an HTTP endpoint.
+
+    This class defines some common attributes and methods for handling HTTP requests.
+
+    Attributes:
+        _allowed_methods (list[str]): A list of allowed HTTP methods for this endpoint.
+    """
+
+    def __init__(self) -> None:
+        """Initialize the base endpoint with the allowed methods."""
+        self._allowed_methods = [
+            method
+            for method in [str(member) for member in HTTPMethod.__members__.values()]
+            if getattr(self, method.lower(), None) is not None
+        ]
+
+    def dispatch(self, request_method: str) -> Any:
+        """Dispatch the request to the appropriate method.
+
+        Args:
+            request_method (str): The request method to dispatch.
+
+        Returns:
+            Any: The result of calling the corresponding method.
+        """
+        if resp := self.method_not_allowed(request_method):
+            return resp
+        return getattr(self, request_method)
+
+    def method_not_allowed(
+        self, scope: str
+    ) -> Callable[..., Coroutine[Any, Any, Response]] | None:
+        """Check if the request method is not allowed and return a not_allowed response.
+
+        Args:
+            scope (str): The request method to check.
+
+        Returns:
+            Union[Callable[..., Coroutine[Any, Any, Response]], None]:
+            A coroutine function that returns a response with status code 405 if the method is
+            not allowed, or None otherwise.
+        """
+        # If we're running inside a starlette application then raise an
+        # exception, so that the configurable exception handler can deal with
+        # returning the response. For plain ASGI apps, just return the response.
+        headers = {"Allow": ", ".join(self._allowed_methods)}
+        if scope.upper() not in self._allowed_methods:
+            return partial(not_allowed, headers=headers)
+        return None
+
+
+class StaticResponse(BaseEndpoint):
+    """A class that represents a static HTML response.
+
+    This class handles GET requests by returning the content of an HTML file.
+
+    Attributes:
+        html_file (HTMLPath): The HTML file path to read from.
+        response_kwargs (dict): The keyword arguments to pass to the response object.
+    """
+
+    def __init__(self, html_file: HTMLPathType, **response_kwargs) -> None:
+        """Initialize the static response with an HTML file and optional keyword arguments.
+
+        Args:
+            html_file (HTMLPathType): The HTML file path to read from.
+            **response_kwargs: The keyword arguments to pass to the response object.
+        """
+        super().__init__()
+        self.html_file = HTMLPath(html_file)
+        self.response_kwargs = response_kwargs
+
+    async def get(self, *args, **kwargs) -> Response:
```

Comment:
I'm confused here
you only have GET, how do you handle POST and ETC? do you actually serve it with a GET request? with if request has body?

Reply:
we write this class for handle html files
```python
class StaticResponse(BaseEndpoint):
    """A class that represents a static HTML response...."""
``` 
if the name isn't meaningful so we most change the name.

we can POST or etc methods in html file?

Reply:
We changed this part to handle also JSON files, so we able to use all http methods to this class :)